openlibrary/solr/updater/author.py

```diff
@@ -85,8 +102,39 @@ def work_count(self) -> int:
     @property
     def top_subjects(self) -> list[str]:
         all_subjects = []
-        for counts in self._solr_reply['facet_counts']['facet_fields'].values():
-            for s, num in counts:
-                all_subjects.append((num, s))
+        for counts in self._solr_reply['facets']:
+            if isinstance(counts, dict):
+                buckets = counts.get("buckets") or {}
+                for items in buckets:
+                    all_subjects.append((items.count, items.val))
         all_subjects.sort(reverse=True)
         return [s for num, s in all_subjects[:10]]
+
+    def build(self) -> SolrDocument:
+        doc = cast(dict, super().build())
+        doc |= self.build_ratings() or {}
+        doc |= self.build_reading_log() or {}
+        return cast(SolrDocument, doc)
+
+    def build_ratings(self) -> WorkRatingsSummary:
+        return Ratings.work_ratings_summary_from_counts(
+            [
+                self._solr_reply["facets"].get("ratings_count_%s" % str(index))
+                for index in range(1, 6)
+            ]
+        )
+
+    def build_reading_log(self) -> WorkReadingLogSolrSummary:
+        reading_log = {
+            "want_to_read_count": self._solr_reply["facets"].get("want_to_read_count")
```

Comment:
Can we do this instead? Or is it set to `None` ?
```suggestion
            "want_to_read_count": self._solr_reply["facets"].get("want_to_read_count", 0.0)
```

Reply:
0.0 Seems to work. Resolved.